### PR TITLE
Add playPickupItemAnimation to LivingEntity

### LINKED ISSUE #1824: Add a LivingEntity#playPickupItemAnimation method
Inside the nms class EntityLiving there is a method, EntityLiving#receive that accepts an entity and an integer. This method plays the pickup animation for that specific LivingEntity to any nearby player if the entity is a dropped item, an arrow or an experience orb. The integer is the amount of a dropped item(For the itemstack) or 1 in case of others. I am not sure what it does, maybe just have the animation show different model based on the quantity.

Would be cool if this method can be somehow exposed to allow that specific animation to be played.

### PR DESCRIPTION
**Pull request by Proximyst**
Updated #1966.

### CODE REVIEW

File: Spigot-Server-Patches/0559-Add-playPickupItemAnimation-to-LivingEntity.patch [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+index 3777aba5f32f58f4620d4fe496af4e641ff8c858..e857d75a539e8bd938cb732f9a9faa015e367d08 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+@@ -2,42 +2,7 @@ package org.bukkit.craftbukkit.entity;
+ 
+ import com.google.common.base.Preconditions;
+ import com.google.common.collect.Sets;
+-import java.util.ArrayList;
```
Comment (MiniDigger) [inline Spigot-Server-Patches/0559-Add-playPickupItemAnimation-to-LivingEntity.patch]: E V I L

**Review by MiniDigger — CHANGES_REQUESTED** [review]
just the imports, also not sure about the name, I would call it playItemPickupAnimation but I also speak minilang soooo

File: Spigot-Server-Patches/0559-Add-playPickupItemAnimation-to-LivingEntity.patch (hunk 1 above)
Comment (Proximyst) [reply Spigot-Server-Patches/0559-Add-playPickupItemAnimation-to-LivingEntity.patch]: IntelliJ goes rogue, part 37169

Comment (kennytv) [reply Spigot-Server-Patches/0559-Add-playPickupItemAnimation-to-LivingEntity.patch]: _always blame it on the ide_

**Comment by Proximyst** [comment]
I'm not so sure about the method name; as it stands now it's pretty okay in my non-minilang eyes, but I am not a native English speaker either.

**Comment by kashike** [comment]
It is worth noting that this will not only play the animation, but the item entity will be removed on the client-side.

**Review by MiniDigger — DISMISSED** [review]

**Review by aikar — DISMISSED** [review]

**Comment by Proximyst** [comment]
@limbo-app ready

**Review by limbo-app — APPROVED** [review]
Pull request automatically approved for merging - a core team member has previously approved this pull request.

**Comment by rrvk** [comment]
Should this not be merged to the master instead of the progres branch?

**Comment by Proximyst** [comment]
It should and will; didn't notice the base it was on. We're going to have to handle all the PRs based on that branch...

**Pull request merged**